src/processing/BoardState.java

```diff
@@ -0,0 +1,80 @@
+package processing;
+
+import java.util.*;
+import java.io.Serializable;
+import processing.utility.*;
+import processing.board_object.*;
+
+/**
+* This BoardState class stores two maps which should be serializable for persistance.
+* ClientBoardState and ServerBoardState will store the object of this class in their class.
+*
+* @author Himanshu Jain
+*/
+
+public class BoardState implements Serializable {
+
+	//To authenticate that the user deserializing this object.
+	private static final long serialVersionUID = 1L;
+	
+	 //Map from object ID to the Board Object
+	private Map <ObjectId, BoardObject> objIdToBoardObject;
+	
+	// Map from position on the board to Priority of objects on that position
+	private Map <Position, PriorityQueue <PriorityQueueObject>> pixelToObjects;
+	
+	//constructor to initialize both the maps
+	public BoardState() {
+		objIdToBoardObject = new HashMap <ObjectId, BoardObject> ();
+		pixelToObjects = new HashMap <Position, PriorityQueue <PriorityQueueObject>> ();
+	}
+	
+	//get the Board Object given objectId
+	public BoardObject getObject(ObjectId objectId) {
```

Comment:
You don't need to implement these functions. @himanshu2047 
I am going to implement them in `SelectDeleteUtil` class. As decided earlier, you are just going to keep the maps in the `BoardState`.

Also, these methods should be `synchronized` as the two maps are common to the entire board and can be accessed by different threads.

Reply:
I have defined these function just to access the maps, rest what to do inside them you will define and also thats why I didn't make these functions synchornized, you will make your functions accessing these maps synchronized.

Reply:
Like getting most probable objectId, also like for insertion I didn't make any function to insert, my insert will need the complete priorityqueue but your insert will take different arguments, so my functions are just the basic necessary function not specific to our project. You will implement specifically for this project.

Reply:
If some of your function is like I have implemented like one is there removeObjectFromMap, then you define that function and inside yours call my delete, I guess only for delete this situation is there.

Reply:
> Like getting most probable objectId, also like for insertion I didn't make any function to insert, my insert will need the complete priorityqueue but your insert will take different arguments, so my functions are just the basic necessary function not specific to our project. You will implement specifically for this project.

then why are you defining them at all?

Reply:
> I have defined these function just to access the maps, rest what to do inside them you will define and also thats why I didn't make these functions synchornized, you will make your functions accessing these maps using my defined functions as synchronized.

I have defined functions to access maps. I don't want our internal modules to have function calls like `insertObject(Pos, PriorityQueueObject)`. First arg should be `ArrayList<Pos>` so that we do not lock and release the locks on the maps repeatedly.

I have thought through these minute details. My portion will reduce significantly if you try to implement these functions.

Reply:
Thats all I did, no more functions I will implement.

Reply:
I also just used maps insert delete inside all the functions, not more than that

Reply:
yeah but the utility class will contain all these helper methods as private.